--- nginx.spec	2017-10-18 08:08:23.000000000 +0000
+++ /tmp/nginx.spec	2018-01-26 18:03:40.583087300 +0000
@@ -26,6 +26,7 @@
 
 Source0:           http://nginx.org/download/nginx-%{version}.tar.gz
 Source1:           http://nginx.org/download/nginx-%{version}.tar.gz.asc
+Source2:           https://github.com/ScaleFT/nginx_auth_accessfabric/archive/v1.0.0.tar.gz
 Source10:          nginx.service
 Source11:          nginx.logrotate
 Source12:          nginx.conf
@@ -86,6 +87,7 @@
 Requires:          nginx-mod-http-xslt-filter = %{epoch}:%{version}-%{release}
 Requires:          nginx-mod-mail = %{epoch}:%{version}-%{release}
 Requires:          nginx-mod-stream = %{epoch}:%{version}-%{release}
+Requires:          nginx-mod-http-auth-accessfabric = %{epoch}:%{version}-%{release}
 
 %description all-modules
 %{summary}.
@@ -168,6 +170,14 @@
 %description mod-stream
 %{summary}.
 
+%package mod-http-auth-accessfabric
+Group:             System Environment/Daemons
+Summary:           ScaleFT Accessfabric Auth module 
+Requires:          nginx
+Requires:          libxjwt-devel
+
+%description mod-http-auth-accessfabric
+%{summary}.
 
 %prep
 %setup -q
@@ -175,6 +185,9 @@
 
 cp %{SOURCE200} .
 cp %{SOURCE210} .
+cp %{SOURCE2} .
+
+tar xvf $RPM_SOURCE_DIR/v1.0.0.tar.gz -C $RPM_BUILD_DIR/
 
 %if 0%{?rhel} < 8
 sed -i -e 's#KillMode=.*#KillMode=process#g' %{SOURCE10}
@@ -238,6 +251,7 @@
 %endif
     --with-debug \
     --with-cc-opt="%{optflags} $(pcre-config --cflags)" \
+    --add-dynamic-module="$RPM_BUILD_DIR/nginx_auth_accessfabric-1.0.0" \
     --with-ld-opt="$RPM_LD_FLAGS -Wl,-E" # so the perl module finds its symbols
 
 make %{?_smp_mflags}
@@ -303,6 +317,8 @@
     > %{buildroot}%{_datadir}/nginx/modules/mod-mail.conf
 echo 'load_module "%{_libdir}/nginx/modules/ngx_stream_module.so";' \
     > %{buildroot}%{_datadir}/nginx/modules/mod-stream.conf
+echo 'load_module "%{_libdir}/nginx/modules/ngx_http_auth_accessfabric_module.so";' \
+    > %{buildroot}%{_datadir}/nginx/modules/mod-http-auth-accessfabric.conf
 
 %pre filesystem
 getent group %{nginx_user} > /dev/null || groupadd -r %{nginx_user}
@@ -344,6 +360,11 @@
     /usr/bin/systemctl reload nginx.service >/dev/null 2>&1 || :
 fi
 
+%post mod-http-auth-accessfabric
+if [ $1 -eq 1 ]; then
+    /usr/bin/systemctl reload nginx.service >/dev/null 2>&1 || :
+fi
+
 %preun
 %systemd_preun nginx.service
 
@@ -428,6 +449,11 @@
 %{_datadir}/nginx/modules/mod-stream.conf
 %{_libdir}/nginx/modules/ngx_stream_module.so
 
+%files mod-http-auth-accessfabric
+%{_datadir}/nginx/modules/mod-http-auth-accessfabric.conf
+%{_libdir}/nginx/modules/ngx_http_auth_accessfabric_module.so
+
+
 
 %changelog
 * Wed Oct 18 2017 Luboš Uhliarik <luhliari@redhat.com> - 1:1.12.2-1
